lib/browser/browser.ts

```diff
@@ -183,6 +206,20 @@ class OktaAuthBrowser extends OktaAuthBase implements OktaAuth, SignoutAPI {
 
     this.emitter = new Emitter();
     this.tokenManager = new TokenManager(this, args.tokenManager);
+
+    // Listen on tokenManager events to update authState
+    this.emitter.on('added', (key, token) => {
+      this.authState = { ...this.authState, [key]: token };
```

Comment:
authState should be constructed in only one place (the `updateAuthState` method)

Reply:
The idea here is to update authState synchronously by events, then the ~~`updateAuthState`~~`emitAuthState` (with cancalable promise) handle how to push events to clients. The fn name should be changed.

Reply:
I think we're passing around a lot of architecture decisions implicitly - can we give a one or two paragraph summary of the major parts and their purpose.  

Example (I think this is what you are saying, but if I'm confused that says why we should have this summary):

- updateAuthState() is called to request that the authState be rebuilt (and should prevent overlapping rebuilds causing a storm of emitted events)
- events other than authStateChange are emitted to indicate the authState should change in only the way indicated.
- the authStateChange is a general event to any change and indicates the receiver should call getAuthState() to get any updates


If that's what you are saying, I'm with Aaron - the events should signal to rebuild the authState, but shouldn't assume the previous authState is correct except for the one change.

Can you summarize your architecture?

Reply:
@swiftone There is some optimization that has been introduced in this implementation (implicitly). 
- update tokens in authState synchronously, which driven by the events from tokenManager.
- trigger cancellable promises (throttle the calls on custom callbacks and events emit to clients) when tokens has been updated in `authState` to evaluate the rest states (isAuthenticated, isPending)
- Downstream SDKs consume the latest authState (provided in the callback) by listening on `authStateChange` event.

Talked with Aaron, now I considering should just lift logic from react-sdk up to make this change safe.

Reply:
@aarongranick-okta Only update state in one place now.